internal/pkg/agent/application/monitoring/liveness.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package monitoring
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+)
+
+const formValueKey = "failon"
+
+type LivenessFailConfig struct {
+	Degraded    bool `yaml:"degraded" config:"degraded"`
+	Failed      bool `yaml:"failed" config:"failed"`
+	Coordinator bool `yaml:"coordinator" config:"coordinator"`
+}
+
+// process the form values we get via HTTP
+func handleFormValues(req *http.Request) (LivenessFailConfig, error) {
+	err := req.ParseForm()
+	if err != nil {
+		return LivenessFailConfig{}, fmt.Errorf("Error parsing form: %w", err)
+	}
+
+	defaultUserCfg := LivenessFailConfig{Degraded: false, Failed: true, Coordinator: true}
```

Comment:
We do not want `Failed` to be on by default. If somebody typos an integration configuration, that can restart the agent rapidly, so rapidly it can't process the configuration change from Fleet to correct the typo.

I also don't think we need `coordinator` as an explicit setting, mostly because no user it going to understand what that means and it is leaking implementation details of the agent.

Degraded and Failed are part of the Fleet API contract. We could rename the coordinator at any time and then this setting name makes no sense.

If you want an explicit name for this mode, call it "heartbeat" or something more generic.

Reply:
Yeah, good point about how `failed` might behave. So, just to be clear, the default behavior should be like this?

- If no `failon` param is specified, _just_ check the coordinator
- if `failon == failed` just check for a `Failed` state
- if `failon == degraded` check for both `Failed` and `Degraded` state

Does that sound right?

Reply:
Yes